### PR TITLE
fix: simplify literal boolean checks

### PR DESCRIPTION
**Pull request by DanielRuf**
### Description (*)

We can typecast to boolean and check the result directly.

### Fixed Issues (if relevant)

1. magento/magento2#<issue_number>: Issue title
2. ...

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @DanielRuf. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

File: app/code/Magento/Analytics/Test/Unit/Model/ExportDataHandlerTest.php [hunk 1]
```diff
@@ -191,7 +191,7 @@ public function testPrepareExportData($isArchiveSourceDirectory)
             ->with(
                 $archiveSource,
                 $archiveAbsolutePath,
-                $isArchiveSourceDirectory ? true : false
+                (bool) $isArchiveSourceDirectory
```
Comment (orlangur) [inline app/code/Magento/Analytics/Test/Unit/Model/ExportDataHandlerTest.php]: It's already boolean, no need in typecasting.

File: app/code/Magento/Analytics/ReportXml/DB/ColumnsResolver.php [hunk 2]
```diff
@@ -76,7 +76,7 @@ public function getColumns(SelectBuilder $selectBuilder, $entityConfig)
             $columnName = $this->nameResolver->getName($attributeData);
             if (isset($attributeData['function'])) {
                 $prefix = '';
-                if (isset($attributeData['distinct']) && $attributeData['distinct'] == true) {
+                if ((bool) $attributeData['distinct']) {
```
Comment (orlangur) [inline app/code/Magento/Analytics/ReportXml/DB/ColumnsResolver.php]: Will throw notice when not set, `(bool)` casting is redundant here.

Correct replacement: `!empty($attributeData['distinct'])`

**Review by orlangur — CHANGES_REQUESTED** [review]

**Comment by magento-engcom-team** [comment]
Hi @DanielRuf
Thank you for your contribution. This pull request has been combined (merged) with https://github.com/magento/magento2/pull/19672 to optimize pull requests queue on magento/magento2 repository
Closing this pull request.

**Pull request closed without merging**